Mage.Verify/src/test/java/mage/verify/VerifyCardDataTest.java

```diff
@@ -2288,18 +2338,30 @@ private void checkMissingAbilities(Card card, MtgJsonCard ref) {
             }
         }
 
-        boolean needTargetedAbility = targetedKeywords.stream().anyMatch(refLowerText::contains);
-        boolean foundTargetedAbility = card.getAbilities()
-                .stream()
-                .map(Ability::getTargets)
-                .flatMap(Collection::stream)
-                .anyMatch(target -> !target.isNotTarget());
-        boolean foundProblem = needTargetedAbility != foundTargetedAbility;
-        if (checkMissTargeted && needTargetedAbility && foundProblem) {
-            fail(card, "abilities", "wrong target settings (must be targeted, but it not)");
-        }
-        if (checkMissNonTargeted && !needTargetedAbility && foundProblem) {
-            fail(card, "abilities", "wrong target settings (must set withNotTarget(true), but it not)");
+        String[] excludedCards = {"Lodestone Bauble"}; // Needs to choose a player before targets are selected
+        if (!Arrays.stream(excludedCards).anyMatch(x -> x.equals(ref.name))) {
+            boolean needTargetedAbility = singularTargetRegexPattern.matcher(refLowerText).find() || pluralTargetsRegexPattern.matcher(refLowerText).find() || targetKeywordRegexPattern.matcher(refLowerText).find();
+            boolean recursiveAbilityRefText = indirectTriggerTargetRegexPattern.matcher(refLowerText).find() || quotedTargetRegexPattern.matcher(refLowerText).find();
+            List<Ability> abilities = card.getAbilities().copy();
+            if (card.getSecondCardFace() != null) {
+                abilities.addAll(card.getSecondCardFace().getAbilities());
+            }
+            List<Target> targets = abilities.stream().flatMap(ability -> ability.getModes().values().stream())
```

Comment:
Please explain here -- why you check whole card with all abilities instead make a per ability check?

Reply:
nvm, it's xmage card, not ref